spring-pulsar/src/main/java/org/springframework/pulsar/core/PulsarOperations.java

```diff
@@ -71,7 +71,20 @@ default MessageId send(T message, MessageRouter messageRouter) throws PulsarClie
 	 * @return the id of the sent message
 	 * @throws PulsarClientException if an error occurs
 	 */
-	MessageId send(String topic, T message, MessageRouter messageRouter) throws PulsarClientException;
+	default MessageId send(String topic, T message, MessageRouter messageRouter) throws PulsarClientException {
+		return send(topic, message, messageRouter, null);
+	}
+
+	/**
+	 * Sends a message to the specified topic in a blocking manner.
+	 * @param topic the topic to send the message to or {@code null} to send to the default topic
+	 * @param message the message to send
+	 * @param messageRouter the optional message router to use
+	 * @param typedMessageBuilderCustomizer the optional TypedMessageBuilder customizer
+	 * @return the id of the sent message
+	 * @throws PulsarClientException if an error occurs
+	 */
+	MessageId send(String topic, T message, MessageRouter messageRouter, TypedMessageBuilderCustomizer<T> typedMessageBuilderCustomizer) throws PulsarClientException;
```

Comment:
Wdyt about adding a few more combos to accommodate use of customizer w/o router (and vice-versa)?

**Current combos:**
```
- message
- message / topic
- message / router 
- message / topic / router
```
**New combos something like this:** 
``` 
- message / customizer
- message / topic / customizer
- message / customizer / router
- message / topic / customizer / router
```
The tradeoff of course is signature explosion vs. user convenience. For me, the concerns of signature explosion is lightened by the `default impls delegate to the single abstract w/ all of the params` pattern we are using.

The other tiny change I think we can make it put the customizer in front of the router in the arg list as I bet router is used less than customizer and the customizer would come directly after the payload arg (cohesive).

Reply:
Thanks for the suggestions, and yes, having the customizer together with the payload makes sense 👍